### PR TITLE
scheduler: make history-sample-interval and history-sample-duration configurable

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7877

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

- Has the configuration change

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/statistics/store_load.go [hunk 1]
```diff
@@ -294,36 +303,46 @@ func (s *StoreHistoryLoads) Get(storeID uint64, rwTp utils.RWType, kind constant
 	return load.get()
 }
 
+// UpdateConfig updates the sample duration and interval.
+func (s *StoreHistoryLoads) UpdateConfig(sampleDuration time.Duration, sampleInterval time.Duration) *StoreHistoryLoads {
+	if s.sampleDuration == sampleDuration && s.sampleInterval == sampleInterval {
+		return s
+	}
+	return NewStoreHistoryLoads(s.dim, sampleDuration, sampleInterval)
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/statistics/store_load.go:311]: I think it's not necessary and it will make code complex

**Comment by HuSharp** [comment]
tested
img_v3_028o_d4df4369-4424-4b69-950a-e552dff3a36g

File: pkg/schedule/schedulers/hot_region_config.go [hunk 2]
```diff
@@ -147,6 +153,9 @@ type hotRegionSchedulerConfig struct {
 	ForbidRWType string `json:"forbid-rw-type,omitempty"`
 	// SplitThresholds is the threshold to split hot region if the first priority flow of on hot region exceeds it.
 	SplitThresholds float64 `json:"split-thresholds"`
+
+	HistorySampleDuration typeutil.Duration `json:"history-sample-duration"`
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/schedule/schedulers/hot_region_config.go:157]: Of course. I will hidden some config in another pr.

**Review by HuSharp — APPROVED** [review]
ci failed :)

File: pkg/statistics/store_load.go [hunk 3]
```diff
@@ -272,20 +277,24 @@ func NewStoreHistoryLoads(dim int) *StoreHistoryLoads {
...
-func (s *StoreHistoryLoads) Add(storeID uint64, rwTp utils.RWType, kind constant.ResourceKind, loads []float64) {
+func (s *StoreHistoryLoads) Add(storeID uint64, rwTp utils.RWType, kind constant.ResourceKind, pointLoad []float64) {
 	load, ok := s.loads[rwTp][kind][storeID]
 	if !ok {
-		size := defaultSize
-		if historySampleInterval != 0 {
-			size = int(HistorySampleDuration / historySampleInterval)
+		size := int(DefaultHistorySampleDuration / DefaultHistorySampleInterval)
+		if s.sampleInterval != 0 {
+			size = int(s.sampleDuration / s.sampleInterval)
```
Comment (HuSharp) [inline pkg/statistics/store_load.go:285]: should we hard code a fixed ratio as size rather than supporting a interval? maybe can reduce some configuration

**Review by rleungx — APPROVED** [review]

**Pull request merged**